### PR TITLE
[DR-1280] Add text to show if the account its closed or not

### PR DESCRIPTION
**Pull request by mvillaseco**
# Overview
 Add text to show if the account it's closed or not.

# Changes
 - Add scope variable
 - Add text.

# In progress
 - Text in spanish

### CODE REVIEW

File: src/wwwroot/locales/es-translation.js [hunk 1]
```diff
@@ -348,5 +348,6 @@
         "mediumDate": "dd MMM, yyyy",
         "short": "dd/MM/yyyy HH:mm:ss"
     }
-  }
+  },
+  "my_plan_account_closed": "Tu cuenta ha sido cancelada o tu cuenta gratis ha expirado"
```
Comment (nreal) [inline src/wwwroot/locales/es-translation.js]: Do you think about this?
`"Tu cuenta ha sido cancelada o ha expirado"`

File: src/wwwroot/partials/settings/my-plan.html [hunk 2]
```diff
@@ -41,7 +41,8 @@
       </div>
       <div class="flex two flex--wrap">
         <label class="width--full special uppercase">{{ 'my_plan_renewal_date' | translate }}</label>
```
Comment (andresmoschini) [inline src/wwwroot/partials/settings/my-plan.html]: I think that this label will also change. There are three different scenarios:

1. When `accountEndDate` has value and _the account is free_ it should say "Free trial end date" (or something similar, please confirm)
2. When `accountEndDate` has value and _the account is NOT free_ it should say "Account end date" (or something similar, please confirm)
3. When `accountEndDate` do not have value, it should say "Renewal date" (as now)

Of course, the value should be related to the label.

File: src/wwwroot/partials/settings/my-plan.html [hunk 3]
```diff
@@ -41,7 +41,8 @@
...
-        <p class="renewal-date">{{ vm.resetDate | date :'mediumDate' :'+0' }}</p>
+        <p class="renewal-date" ng-if="!vm.isAccountClosed">{{ vm.resetDate | date :'mediumDate' :'+0' }}</p>
+        <p class="renewal-date" ng-if="vm.isAccountClosed">{{ 'my_plan_account_closed' | translate }}</p>
```
Comment (andresmoschini) [inline src/wwwroot/partials/settings/my-plan.html]: I think that it should always show a date (see [previous comment](https://github.com/DopplerRelay/relay-webapp/pull/111/files#diff-5f616ca107824eaba403e01898c21b0bR43))

When the trial has ended, it should be shown in another place. Maybe in the first field (`price_text`)?

File: src/wwwroot/controllers/PlanCtrl.js [hunk 4]
```diff
@@ -57,7 +57,25 @@
       return settings.getStatusPlanInfo()
           .then(function (result) {
             vm.extraEmailsSent = 0;
-            vm.resetDate = result.data.endDate;
+            vm.isAccountClosed = result.data.accountClosed;
+            if (result.data.accountEndDate) {
+              vm.resetDate = result.data.accountEndDate;
```
Comment (andresmoschini) [inline src/wwwroot/controllers/PlanCtrl.js]: In my opinion, resetDate should always refer to current period ending (`result.data.endDate`)

File: src/wwwroot/controllers/PlanCtrl.js [hunk 5]
```diff
@@ -57,7 +57,25 @@
...
+              if (vm.isFreeTrial){
+                vm.labelDate = 'my_plan_free_trial_end_label';
+              } else {
+                vm.labelDate = 'my_plan_account_end_label';
```
Comment (andresmoschini) [inline src/wwwroot/controllers/PlanCtrl.js]: I am not comfortable with the idea of share the same label for two different concepts

Comment (andresmoschini) [inline src/wwwroot/controllers/PlanCtrl.js]: Sorry, in this case, it is the same concept (end date) for different situation, I like it.

File: src/wwwroot/controllers/PlanCtrl.js [hunk 6]
```diff
@@ -57,7 +57,25 @@
...
+              }
+            } else if (!result.data.accountEndDate) {
+              vm.labelDate = 'my_plan_renewal_date';
```
Comment (andresmoschini) [inline src/wwwroot/controllers/PlanCtrl.js]: This is what makes me uncomfortable, Does `labelDate` refer to _reset date_ or to _end date_?

File: src/wwwroot/partials/settings/my-plan.html [hunk 7]
```diff
@@ -40,8 +41,8 @@
         <spinner class="spinner" ng-class="vm.planStatusInfoLoader ?  'show':'hide'"></spinner>
       </div>
       <div class="flex two flex--wrap">
-        <label class="width--full special uppercase">{{ 'my_plan_renewal_date' | translate }}</label>
-        <p class="renewal-date">{{ vm.resetDate | date :'mediumDate' :'+0' }}</p>
+        <label class="width--full special uppercase">{{ vm.labelDate | translate }}</label>
+        <p class="renewal-date">{{ vm.resetDate | date :'mediumDate' :'+0' }}</p>
```
Comment (andresmoschini) [inline src/wwwroot/partials/settings/my-plan.html]: Seeing the view I understand your decision, it looks really simple. Anyway, I still prefer a more complex view but having meaningful fields.

**Comment by andresmoschini** [comment]
@mvillaseco did you verify how webapp will work if some fields are not present?

See https://github.com/MakingSense/Doppler_Relay/pull/1042

**Comment by andresmoschini** [comment]
@mvillaseco as I understand, your code is valid and will work fine. My comments are only because I think that meaningful fields will be more maintainable in the future and easier to read now, but it is very openable, so, feel free to continue with your approach.

Do you have a screenshot of an already expired account?

**Comment by mvillaseco** [comment]
Trial: https://www.screencast.com/t/9zVzsxdoF

File: src/wwwroot/locales/es-translation.js [hunk 8]
```diff
@@ -348,5 +348,9 @@
         "mediumDate": "dd MMM, yyyy",
         "short": "dd/MM/yyyy HH:mm:ss"
     }
-  }
+  },
+  "my_plan_free_trial_end_label": "Fecha de prueba gratuita",
+  "my_plan_account_end_label": "Finalización de cuenta",
+  "my_plan_free_trial_closed" : "Tu demo gratuita ha finalizado.",
+  "my_plan_account_closed": "Tu demo gratuita ha finalizado."
```
Comment (andresmoschini) [inline src/wwwroot/locales/es-translation.js]: Remember to update this text before merge.

**Comment by andresmoschini** [comment]
Thanks @mvillaseco, I think that it is a little more verbose but also cleaner than before.

**Pull request merged**